### PR TITLE
added max allowed input resolution AO RTSP

### LINKED ISSUE #464: Add max frame geometry size to AO-RTSP
Fix the maximum possible safe resolution supported by AO-RTSP. Terminate with prompt:

```
The resolution of the incoming stream is treater than the allowed max 'known-working' resolutions WxH. Terminate. You can override the max allowed resolution with 'MAX_ALLOWED_RESOLUTION' environment variable.

### PR DESCRIPTION
**Pull request by dorgun**

### CODE REVIEW

File: gst_plugins/python/savant_rs_video_demux.py [hunk 1]
```diff
@@ -84,6 +84,24 @@
         None,
         GObject.ParamFlags.READWRITE,
     ),
+    'max-allowed-resolution-width': (
```
Comment (tomskikh) [inline gst_plugins/python/savant_rs_video_demux.py]: Names of the properties too long, `max-width`, `max-height` should be enough.

File: docs/source/savant_101/10_adapters.rst [hunk 2]
```diff
@@ -833,6 +833,7 @@ The simplified design of the adapter is depicted in the following diagram:
 - ``SYNC_OUTPUT``: a flag indicates whether to show frames on sink synchronously (i.e. at the source rate); the streaming may be not stable with this flag, try to avoid it; the default value is ``False``;
 - ``SOURCE_ID``: a filter to receive frames with a specific ``source_id`` only. The sink works in single-stream mode when this option is specified;
 - ``SOURCE_IDS``: a filter to receive frames with specific ``source_id``-s only. The sink works in multi-stream mode when this option is specified.
+- ``MAX_ALLOWED_RESOLUTION``: Maximum resolution of the incoming stream. If the resolution is greater than the allowed resolution, the adapter will terminate. You can override the max allowed resolution be setting width and height of frames. The default value is ``3840x2152``.
```
Comment (tomskikh) [inline docs/source/savant_101/10_adapters.rst]: The adapter won't terminate itself. I don't think it should. By design the adapter always works and shows either video stream or stub image.
In this case the adapter will show an error, terminate and restart the input pipeline.

Comment (tomskikh) [reply docs/source/savant_101/10_adapters.rst]:
```
 ERROR insight::savant::savant_rs_video_demux                       > The resolution of the incoming stream is 1280x720 and treater than the allowed max 100x100 resolutions. Terminate. You can override the max allowed resolution with 'MAX_ALLOWED_RESOLUTION' environment variable.
 ERROR insight::savant::gstreamer::runner                           > Received error "gst-stream-error-quark: The resolution of the incoming stream is 1280x720 and treater than the allowed max 100x100 resolutions. Terminate. You can override the max allowed resolution with 'MAX_ALLOWED_RESOLUTION' environment variable. (9)" from savant_rs_video_demux+savantrsvideodemux6.
Exception in thread Thread-22:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/savant/gstreamer/runner.py", line 75, in _main_loop_run
    raise RuntimeError(self._error)
RuntimeError: Received error "gst-stream-error-quark: The resolution of the incoming stream is 1280x720 and treater than the allowed max 100x100 resolutions. Terminate. You can override the max allowed resolution with 'MAX_ALLOWED_RESOLUTION' environment variable. (9)" from savant_rs_video_demux+savantrsvideodemux6. Debug info: "/opt/savant/gst_plugins/python/savant_rs_video_demux.py(561): check_max_allowed_resolution (): /GstPipeline:input-pipeline/savant_rs_video_demux+SavantRsVideoDemux:savant_rs_video_demux+savantrsvideodemux6".
```

File: adapters/ds/sinks/always_on_rtsp/config.py [hunk 3]
```diff
@@ -66,6 +66,17 @@ def __init__(self):
 
         self.framerate = opt_config('FRAMERATE', '30/1')
         self.sync = opt_config('SYNC_OUTPUT', False, strtobool)
+        self.max_allowed_resolution = opt_config(
+            'MAX_ALLOWED_RESOLUTION',
```
Comment (tomskikh) [inline adapters/ds/sinks/always_on_rtsp/config.py]: Do we need a word "allowed" here? `MAX_RESOLUTION` should be enough.

**Review by tomskikh — CHANGES_REQUESTED** [review]

File: docs/source/savant_101/10_adapters.rst (hunk 2 above)
Comment (dorgun) [reply docs/source/savant_101/10_adapters.rst]: @bwsw Should we terminate the adapter or interrupt the display of the input video stream but show the stub image?

File: adapters/ds/sinks/always_on_rtsp/config.py (hunk 3 above)
Comment (dorgun) [reply adapters/ds/sinks/always_on_rtsp/config.py]: Set the title according to the requirements in the task.

Comment (bwsw) [reply adapters/ds/sinks/always_on_rtsp/config.py]: @dorgun please change as Pavel proposes. It is a good one.

File: gst_plugins/python/savant_rs_video_demux.py (hunk 1 above)
Comment (dorgun) [reply gst_plugins/python/savant_rs_video_demux.py]: Fixed it.

File: docs/source/savant_101/10_adapters.rst (hunk 2 above)
Comment (dorgun) [reply docs/source/savant_101/10_adapters.rst]: Modified

File: adapters/ds/sinks/always_on_rtsp/config.py (hunk 3 above)
Comment (dorgun) [reply adapters/ds/sinks/always_on_rtsp/config.py]: Fixed it.

**Review by tomskikh — APPROVED** [review]

**Pull request merged**